### PR TITLE
Append ordinal numbers to test steps names

### PR DESCRIPTION
**Pull request by tosi3k**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
If test configs have steps with duplicate names, they will be erroneously shown as one in the Prow GCS viewer. The change will prevent from such situations as we add ordinal numbers at the end of the test steps' names.

**Which issue(s) this PR fixes**:
No issue is opened for that.

**Special notes for your reviewer**:
/assign wojtek-t

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: clusterloader2/pkg/test/simple_test_executor.go [hunk 1]
```diff
@@ -119,9 +119,11 @@ func (ste *simpleExecutor) ExecuteTestSteps(ctx Context, conf *api.Config) *erro
 
 	errList := errors.NewErrorList()
 	for i := range conf.Steps {
-		if conf.Steps[i].Name == "" {
-			conf.Steps[i].Name = fmt.Sprintf("Step number %d [autogenerated, please name your step in the test config]", i+1)
+		namePrefix := conf.Steps[i].Name
+		if namePrefix == "" {
+			namePrefix = "[autogenerated, please name your step in the test config]"
 		}
+		conf.Steps[i].Name = fmt.Sprintf("%s (step no. %d)", namePrefix, i+1)
```
Comment (wojtek-t) [inline clusterloader2/pkg/test/simple_test_executor.go:126]: nit: [step: %d]

Comment (tosi3k) [reply clusterloader2/pkg/test/simple_test_executor.go]: Done.

**Comment by mm4tt** [comment]
Nice, but I'd prefer if the step number was the prefix of the step name. By doing so, we'd have the tests displayed nicely in the test_grid. Currently they are ordered alphabetically, which is odd 
8EzZbHYPbsqCHJK

**Pull request merged**